core/server/web/parent-app.js

```diff
@@ -17,6 +17,7 @@ module.exports = function setupParentApp(options = {}) {
     parentApp.enable('trust proxy');
 
     parentApp.use(shared.middlewares.logRequest);
+    parentApp.use(shared.middlewares.emitEvents);
```

Comment:
IMO this middleware needs to be registered for the admin API only?

Reply:
With the current case yes, thought behind this here was as its a more generic place to listen to and emit any req/res based events. Guess can move this to admin api now and we can move it back if anything comes later :)